### ISSUE TITLE
OpenAPI to cadl command completes without generating any files and with no errors, for any input.

### ISSUE THREAD

**Issue by kuchatur-ms** [Azure/autorest#4698]
**Before filling a bug**

- [x] have you checked the [faq](https://github.com/Azure/autorest/tree/main/docs/faq.md) for known issues.
- [x] have you checked existing issues

**Describe the bug**
While running the command to generate cadl files from OpenAPI swagger or using readme files doesnt generate any cadl files.
The command completes without any errors, but 0 files are generated. I have tried this with my swagger file and even with the files in the examples/ tests in this repo, with the same result.
**autorest --openapi-to-cadl --input-file=complex-body.json -namespace=complexbodykuchatur --title="complexkuchatur" --use=@autorest/openapi-to-cadl@next**

```
Please provide as much information as you can. This would be:
- OpenAPI files having the issues
- Autorest command used
```
C:\Users\........\swaggertocadl>**autorest --info**
AutoRest code generation utility [cli version: 3.6.3; node: v18.13.0]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
Showing All Installed Extensions

 Type       Extension Name                           Version      Location
 core       @autorest/core                           3.0.6282     C:\Users\kuchatur\.autorest\@autorest_core@3.0.6282
 core       @autorest/core                           3.9.4        C:\Users\kuchatur\.autorest\@autorest_core@3.9.4
 extension  @autorest/modelerfour                    4.26.0       C:\Users\kuchatur\.autorest\@autorest_modelerfour@4.26.0
 extension  @autorest/openapi-to-cadl                0.2.0        C:\Users\kuchatur\.autorest\@autorest_openapi-to-cadl@0.2.0
 extension  @autorest/openapi-to-cadl                0.2.1-dev.1  C:\Users\kuchatur\.autorest\@autorest_openapi-to-cadl@0.2.1-dev.1
 extension  @microsoft.azure/autorest.csharp         2.3.84       C:\Users\kuchatur\.autorest\@microsoft.azure_autorest.csharp@2.3.84
 extension  @microsoft.azure/autorest.modeler        2.3.55       C:\Users\kuchatur\.autorest\@microsoft.azure_autorest.modeler@2.3.55

command result:
C:\Users\........\swaggertocadl>**autorest --openapi-to-cadl --input-file=complex-body.json -namespace=complexbodykuchatur --title="complexkuchatur" --use=@autorest/openapi-to-cadl@next --verbose**
AutoRest code generation utility [cli version: 3.6.3; node: v18.13.0]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
info    | AutoRest core version selected from configuration: ^3.7.0.
info    |    Loading AutoRest core      'C:\Users\kuchatur\.autorest\@autorestcore@3.9.4\nodemodules\@autorest\core\dist' (3.9.4)
verbose | [2.05 s] No configuration found at 'file:///C:/Users/kuchatur/Desktop/az/azure-powershell/src/Migrate/swaggertocadl/-namespace=complexbodykuchatur'.
verbose | [2.08 s]    Including configuration file 'file:///C:/Users/kuchatur/.autorest/@autorestcore@3.9.4/nodemodules/@autorest/core/dist/resources/default-configuration.md'
verbose | [2.16 s]    Including configuration file 'file:///C:/Users/kuchatur/.autorest/@autorestcore@3.9.4/nodemodules/@autorest/core/dist/resources/inspect.md'
verbose | [2.18 s]    Including configuration file 'file:///C:/Users/kuchatur/.autorest/@autorestcore@3.9.4/nodemodules/@autorest/core/dist/resources/directives.md'
verbose | [2.22 s]    Including configuration file 'file:///C:/Users/kuchatur/.autorest/@autorestcore@3.9.4/nodemodules/@autorest/core/dist/resources/pipeline.md'
verbose | [2.25 s]    Including configuration file 'file:///C:/Users/kuchatur/.autorest/@autorestcore@3.9.4/nodemodules/@autorest/core/dist/resources/loader-openapi.md'
verbose | [2.29 s]    Including configuration file 'file:///C:/Users/kuchatur/.autorest/@autorestcore@3.9.4/nodemodules/@autorest/core/dist/resources/miscellaneous.md'
verbose | [2.32 s]    Including configuration file 'file:///C:/Users/kuchatur/.autorest/@autorestcore@3.9.4/nodemodules/@autorest/core/dist/resources/plugin-powershell.md'
verbose | [2.35 s]    Including configuration file 'file:///C:/Users/kuchatur/.autorest/@autorestcore@3.9.4/nodemodules/@autorest/core/dist/resources/plugin-azureresourceschema.md'
verbose | [2.40 s]    Including configuration file 'file:///C:/Users/kuchatur/.autorest/@autorestcore@3.9.4/nodemodules/@autorest/core/dist/resources/plugin-csharp.md'
verbose | [2.46 s]    Including configuration file 'file:///C:/Users/kuchatur/.autorest/@autorestcore@3.9.4/nodemodules/@autorest/core/dist/resources/plugin-go.md'
verbose | [2.51 s]    Including configuration file 'file:///C:/Users/kuchatur/.autorest/@autorestcore@3.9.4/nodemodules/@autorest/core/dist/resources/plugin-java.md'
verbose | [2.58 s]    Including configuration file 'file:///C:/Users/kuchatur/.autorest/@autorestcore@3.9.4/nodemodules/@autorest/core/dist/resources/plugin-nodejs.md'
verbose | [2.64 s]    Including configuration file 'file:///C:/Users/kuchatur/.autorest/@autorestcore@3.9.4/nodemodules/@autorest/core/dist/resources/plugin-php.md'
verbose | [2.70 s]    Including configuration file 'file:///C:/Users/kuchatur/.autorest/@autorestcore@3.9.4/nodemodules/@autorest/core/dist/resources/plugin-python.md'
verbose | [2.74 s]    Including configuration file 'file:///C:/Users/kuchatur/.autorest/@autorestcore@3.9.4/nodemodules/@autorest/core/dist/resources/plugin-ruby.md'
verbose | [2.79 s]    Including configuration file 'file:///C:/Users/kuchatur/.autorest/@autorestcore@3.9.4/nodemodules/@autorest/core/dist/resources/plugin-typescript.md'
verbose | [2.83 s]    Including configuration file 'file:///C:/Users/kuchatur/.autorest/@autorestcore@3.9.4/nodemodules/@autorest/core/dist/resources/plugin-validators.md'
verbose | [2.89 s]    Including configuration file 'file:///C:/Users/kuchatur/.autorest/@autorestcore@3.9.4/nodemodules/@autorest/core/dist/resources/plugin-az.md'
verbose | [2.95 s]    Including configuration file 'file:///C:/Users/kuchatur/.autorest/@autorestcore@3.9.4/nodemodules/@autorest/core/dist/resources/plugin-terraform.md'
verbose | [3.01 s]    Including configuration file 'file:///C:/Users/kuchatur/.autorest/@autorestcore@3.9.4/nodemodules/@autorest/core/dist/resources/plugin-azure-functions.md'
verbose | [3.07 s]    Including configuration file 'file:///C:/Users/kuchatur/.autorest/@autorestcore@3.9.4/nodemodules/@autorest/core/dist/resources/plugin-openapi-to-cadl.md'
verbose | [3.13 s]    Including configuration file 'file:///C:/Users/kuchatur/.autorest/@autorestcore@3.9.4/nodemodules/@autorest/core/dist/resources/graphs.md'
verbose | [3.20 s]    Including configuration file 'file:///C:/Users/kuchatur/.autorest/@autorestcore@3.9.4/nodemodules/@autorest/core/dist/resources/help-configuration.md'
info    |    Loading AutoRest extension '@autorest/openapi-to-cadl' (next->0.2.1-dev.1)
verbose | [3.47 s]    Including extension configuration file 'file:///C:/Users/kuchatur/.autorest/@autorestopenapi-to-cadl@0.2.1-dev.1/nodemodules/@autorest/openapi-to-cadl/README.md'
verbose | [3.47 s]    Including configuration file 'file:///C:/Users/kuchatur/.autorest/@autorestopenapi-to-cadl@0.2.1-dev.1/nodemodules/@autorest/openapi-to-cadl/README.md'
info    |    Loading AutoRest extension '@autorest/modelerfour' (^4.23.5->4.26.0)
verbose | [3.70 s]    Including extension configuration file 'file:///C:/Users/kuchatur/.autorest/@autorestmodelerfour@4.26.0/nodemodules/@autorest/modelerfour/readme.md'
verbose | [3.70 s]    Including configuration file 'file:///C:/Users/kuchatur/.autorest/@autorestmodelerfour@4.26.0/nodemodules/@autorest/modelerfour/readme.md'
verbose | [5.58 s] Reading OpenAPI 2.0 file file:///C:/Users/kuchatur/Desktop/az/azure-powershell/src/Migrate/swaggertocadl/complex-body.json
warning | PreCheck/SchemaMissingType | The schema 'DotFishMarket' with an undefined type and declared properties is a bit ambiguous. This has been auto-corrected to 'type:object'
    - file:///C:/Users/kuchatur/Desktop/az/azure-powershell/src/Migrate/swaggertocadl/complex-body.json:1477:7
warning | PreCheck/CheckDuplicateSchemas | Checking for duplicate schemas, this could take a (long) while.  Run with --verbose for more detail.
verbose | [7.96 s] Found and removed 0 duplicate schema
verbose | [8.78 s]   ModelerFour/api-version-mode:client
verbose | [8.78 s]   ModelerFour/api-version-parameter:constant
verbose | [8.78 s]   ModelerFour/use-model-namespace:false
info    | transformOperationGroup: Transforming object Basic
info    | getDiscriminatorProperty: Transforming property id of type integer
info    | getDiscriminatorProperty: Transforming property name of type string
info    | getDiscriminatorProperty: Transforming property color of type choice
info    | transformOperationGroup: Transforming object Error
info    | getDiscriminatorProperty: Transforming property status of type integer
info    | getDiscriminatorProperty: Transforming property message of type string
info    | transformOperationGroup: Transforming object IntWrapper
info    | getDiscriminatorProperty: Transforming property field1 of type integer
info    | getDiscriminatorProperty: Transforming property field2 of type integer
info    | transformOperationGroup: Transforming object LongWrapper
info    | getDiscriminatorProperty: Transforming property field1 of type integer
info    | getDiscriminatorProperty: Transforming property field2 of type integer
info    | transformOperationGroup: Transforming object FloatWrapper
info    | getDiscriminatorProperty: Transforming property field1 of type number
info    | getDiscriminatorProperty: Transforming property field2 of type number
info    | transformOperationGroup: Transforming object DoubleWrapper
info    | getDiscriminatorProperty: Transforming property field1 of type number
info    | getDiscriminatorProperty: Transforming property field56ZerosAfterTheDotAndNegativeZeroBeforeDotAndThisIsALongFieldNameOnPurpose of type number
info    | transformOperationGroup: Transforming object BooleanWrapper
info    | getDiscriminatorProperty: Transforming property fieldTrue of type boolean
info    | getDiscriminatorProperty: Transforming property fieldFalse of type boolean
info    | transformOperationGroup: Transforming object StringWrapper
info    | getDiscriminatorProperty: Transforming property field of type string
info    | getDiscriminatorProperty: Transforming property empty of type string
info    | getDiscriminatorProperty: Transforming property null of type string
info    | transformOperationGroup: Transforming object DateWrapper
info    | getDiscriminatorProperty: Transforming property field of type date
info    | getDiscriminatorProperty: Transforming property leap of type date
info    | transformOperationGroup: Transforming object DatetimeWrapper
info    | getDiscriminatorProperty: Transforming property field of type date-time
info    | getDiscriminatorProperty: Transforming property now of type date-time
info    | transformOperationGroup: Transforming object Datetimerfc1123Wrapper
info    | getDiscriminatorProperty: Transforming property field of type date-time
info    | getDiscriminatorProperty: Transforming property now of type date-time
info    | transformOperationGroup: Transforming object DurationWrapper
info    | getDiscriminatorProperty: Transforming property field of type duration
info    | transformOperationGroup: Transforming object ByteWrapper
info    | getDiscriminatorProperty: Transforming property field of type byte-array
info    | transformOperationGroup: Transforming object ArrayWrapper
info    | getDiscriminatorProperty: Transforming property array of type array
info    | transformOperationGroup: Transforming object DictionaryWrapper
info    | getDiscriminatorProperty: Transforming property defaultProgram of type dictionary
info    | transformOperationGroup: Transforming object Siamese
info    | getDiscriminatorProperty: Transforming property breed of type string
info    | transformOperationGroup: Transforming object Cat
info    | getDiscriminatorProperty: Transforming property color of type string
info    | getDiscriminatorProperty: Transforming property hates of type array
info    | transformOperationGroup: Transforming object Dog
info    | getDiscriminatorProperty: Transforming property food of type string
info    | transformOperationGroup: Transforming object Pet
info    | getDiscriminatorProperty: Transforming property id of type integer
info    | getDiscriminatorProperty: Transforming property name of type string
info    | transformOperationGroup: Transforming object Fish
info    | getDiscriminatorProperty: Transforming property species of type string
info    | getDiscriminatorProperty: Transforming property length of type number
info    | getDiscriminatorProperty: Transforming property siblings of type array
info    | transformOperationGroup: Transforming object DotFish
info    | getDiscriminatorProperty: Transforming property species of type string
info    | transformOperationGroup: Transforming object DotFishMarket
info    | transformOperationGroup: Transforming object DotSalmon
info    | getDiscriminatorProperty: Transforming property location of type string
info    | getDiscriminatorProperty: Transforming property iswild of type boolean
info    | getDiscriminatorProperty: Getting discriminator property for DotSalmon
info    | getDiscriminatorProperty: Getting discriminator property for DotFish
info    | getDiscriminatorProperty: Transforming property salmons of type array
info    | getDiscriminatorProperty: Transforming property fishes of type array
info    | transformOperationGroup: Transforming object Salmon
info    | getDiscriminatorProperty: Transforming property location of type string
info    | getDiscriminatorProperty: Transforming property iswild of type boolean
info    | transformOperationGroup: Transforming object ReadonlyObj
info    | getDiscriminatorProperty: Transforming property id of type string
info    | getDiscriminatorProperty: Transforming property size of type integer
info    | transformOperationGroup: Transforming object MyBaseType
info    | getDiscriminatorProperty: Transforming property propB1 of type string
info    | transformOperationGroup: Transforming object MyBaseHelperType
info    | getDiscriminatorProperty: Transforming property propBH1 of type string
info    | transformOperationGroup: Transforming object SmartSalmon
info    | getDiscriminatorProperty: Transforming property collegeDegree of type string
info    | getDiscriminatorProperty: Getting discriminator property for SmartSalmon
info    | getDiscriminatorProperty: Getting discriminator property for Salmon
info    | transformOperationGroup: Transforming object Shark
info    | getDiscriminatorProperty: Transforming property age of type integer
info    | getDiscriminatorProperty: Transforming property birthday of type date-time
info    | transformOperationGroup: Transforming object Sawshark
info    | getDiscriminatorProperty: Transforming property picture of type byte-array
info    | getDiscriminatorProperty: Getting discriminator property for Sawshark
info    | getDiscriminatorProperty: Getting discriminator property for Shark
info    | transformOperationGroup: Transforming object Goblinshark
info    | getDiscriminatorProperty: Transforming property jawsize of type integer
info    | getDiscriminatorProperty: Transforming property color of type choice
info    | getDiscriminatorProperty: Getting discriminator property for Goblinshark
info    | getDiscriminatorProperty: Getting discriminator property for Shark
info    | transformOperationGroup: Transforming object Cookiecuttershark
info    | getDiscriminatorProperty: Getting discriminator property for Cookiecuttershark
info    | getDiscriminatorProperty: Getting discriminator property for Shark
info    | transformOperationGroup: Transforming object MyDerivedType
info    | getDiscriminatorProperty: Transforming property propD1 of type string
info    | getDiscriminatorProperty: Getting discriminator property for MyDerivedType
info    | getDiscriminatorProperty: Getting discriminator property for MyBaseType
**info    | Autorest completed in 11.06s. 0 files generated.**

**Expected behavior**

The command should generate the cadl files.

**Additional context**

**Comment by kuchatur-ms** [comment]
Jose suggested using --output-folder=. parameter, but that didnt help. another member from our team was also able to repro this issue.

**Comment by tjprescott** [comment]
@joheredi I did some troubleshooting and can confirm this behavior. 
- I can generate Python and Typescript successfully but not openapi-to-cadl
- I tried all released versions of openapi-to-cadl with `--use` and they all do the same thing
- `autorest --reset` did not have an impact
- results were the same whether I used a README.md or not
- the plugin is clearly running as evidenced by the info statements

**Comment by tjprescott** [comment]
Actually, the following worked for me. While it said that 0 files were generated, the correct files were actually generated:
`autorest --openapi-to-cadl --use=@autorest/openapi-to-cadl@next --require=README.md --input-file=SwaggerViacode.json --output-folder=.`

This *only* worked if `output-folder` was set to `.`

**Comment by joheredi** [comment]
Thanks @tjprescott, I also verified that the command you suggested works around the issue. I have submitted a fix for the issue in https://github.com/Azure/autorest/pull/4702

**Comment by kuchatur-ms** [comment]
I tried the output-folder parameter again. It does work. I might have missed the generated files earlier. I have some other questions though.

1. I didnt get what you mean by this. _"It looks like you are trying to convert a Swagger from the autorest test server. This openapi to cadl plugin is only intended for service descriptions. That may be why you get no output."_. Any swagger files which is openAPI2 format should work here right? 
2. What is the require-readme param used for? If I want to take a swagger we have already published and move it to cadl, which command of the two in the readme we should use? Eg: https://github.com/Azure/azure-rest-api-specs/blob/main/specification/migrate/resource-manager/Microsoft.Migrate/stable/2019-10-01/migrate.json
3. Is there any difference between using readme with a link to the published swagger or giving the swagger as input?
